src/ParseError.js

```diff
@@ -3,25 +3,43 @@
  * functions when something has gone wrong. This is used to distinguish internal
  * errors from errors in the expression that the user provided.
  */
-function ParseError(message, lexer, position) {
+function ParseError(message, token) {
     var error = "KaTeX parse error: " + message;
+    var start;
+    var end;
 
-    if (lexer !== undefined && position !== undefined) {
+    if (token && token.lexer) {
```

Comment:
Yes. Unless I'm missing something, I'm not *actually* constructing tokens without a lexer reference. But I envision that there *might* be situations where it makes sense to construct artificial tokens which don't actually occur in source.

Furthermore, the parse node which [results from a `parseGroup` call](https://github.com/Khan/KaTeX/pull/493/files/e3dda607afd87a8e6b62fc4135a8f4d50cbee697#diff-a0c375bdbb580536aa70722d5d376137L676) will also behave like a token, in that [it will contain `start`, `end` and `lexer` properties](https://github.com/Khan/KaTeX/pull/493/files/e3dda607afd87a8e6b62fc4135a8f4d50cbee697#diff-1e3e2449fa5a2f1a38c89076216368ceR8). And [some](https://github.com/Khan/KaTeX/pull/493/files/e3dda607afd87a8e6b62fc4135a8f4d50cbee697#diff-af9da43267175de02306a3e75d6c1e88R107) [calls](https://github.com/Khan/KaTeX/pull/493/files/e3dda607afd87a8e6b62fc4135a8f4d50cbee697#diff-0ce252df8a9f3b1e09f8552116f0abbeR580) to the `ParseError` constructor pass in a parse node, not a raw token, so I'm not perfectly confident that every group that might end up in such a setup will actually contain the required source position information.

Do you think I should name the argument differently, since it's not only tokens? If so, would `tokenOrNode` or `pos` be a better name?